### PR TITLE
Created 3 endpoints `index`, `new`, `create` for `student_identifier_types`

### PR DESCRIPTION
**Pull request by dinever**
This is part of #614.

In this PR I created 3 endpoints `index`, `new`, `create` for `student_identifier_types`.

After this being merged, I will open another PR for 3 more endpoints: `edit`, `update` and `destroy`.

Everything is hidden under Flipper `student_identifier`.

Showcase:

demo

screen shot 2016-07-08 at 6 56 39 pm

screen shot 2016-07-08 at 6 56 53 pm

screen shot 2016-07-08 at 6 56 59 pm

/cc @johndbritton @tarebyte

### CODE REVIEW

**Comment by tarebyte** [comment]
> Everything is hidden under Flipper `student_identifier`.

🎉

File: app/controllers/student_identifier_types_controller.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    params
+      .require(:student_identifier_type)
+      .permit(:name, :description, :content_type)
+      .merge(organization: @organization)
+  end
+
+  private
+
+  def check_flipper
+    not_found unless Classroom.flipper[:student_identifier].enabled? current_user
```
Comment (tarebyte) [inline app/controllers/student_identifier_types_controller.rb]: Can we wrap `Classroom.flipper[:student_identifier].enabled? current_user` into a helper method on the `ApplicationController`?

Something like

```ruby
def student_identifier_enabled?
  Classroom.flipper[:student_identifier].enabled?(current_user)
end
```

File: spec/controllers/student_identifier_types_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  let(:organization)  { GitHubFactory.create_owner_classroom_org }
+  let(:user)          { organization.users.first                 }
+  let(:student)       { GitHubFactory.create_classroom_student   }
+
+  let(:student_identifier_type) { GitHubFactory.create_student_identifier(organization) }
+
+  before do
+    sign_in(user)
+    Classroom.flipper[:student_identifier].enable
+  end
```
Comment (tarebyte) [inline spec/controllers/student_identifier_types_controller_spec.rb]: Can we add a small set of tests, to ensure that we are returning a `404` if the flag is not on?

File: app/controllers/student_identifier_types_controller.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  def new
+    @student_identifier_type = StudentIdentifierType.new
+  end
+
+  def create
+    @student_identifier_type = StudentIdentifierType.new(student_identifier_type_params)
+    if @student_identifier_type.save
+      flash[:success] = "\"#{student_identifier_type.name}\" has been created!"
+      redirect_to session.delete(:return_to) || { action: 'index' }
```
Comment (tarebyte) [inline app/controllers/student_identifier_types_controller.rb]: Just out of curiosity can you describe the scenario where we set a `:return_to` for the user?

**Comment by dinever** [comment]
Hi @tarebyte, thanks for reviewing my code, all changes mentioned above have been done!

Changes:

- [x] Add a small set of tests, to ensure that we are returning a 404 if the flag is not on: b1e399b7da900fa120c87414add7d2f373ef4842
- [x] Wrap `Classroom.flipper[:student_identifier].enabled? current_user` into a helper method on the `ApplicationController`: a97cfaf66465d51fc123576c471e24cbfffd742a and 70596b0cb24dbead9df77d929612371b25743d53
- [x] Describe the scenario where we set a :return_to for the user: 423c8f1598f9d3a9d507fdcff932bcde37ba74b0

It'd be great if you can give me some feedback on these changes, especially on 423c8f1598f9d3a9d507fdcff932bcde37ba74b0. Thanks! 😄

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -79,6 +79,10 @@ def logged_in?
     !current_user.nil?
   end
 
+  def student_identifier_enabled?
+    Classroom.flipper[:student_identifier].enabled?(current_user)
+  end
```
Comment (tarebyte) [inline app/controllers/application_controller.rb]: 👍

File: app/controllers/student_identifier_types_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+class StudentIdentifierTypesController < ApplicationController
+  include OrganizationAuthorization
+
+  before_action :check_flipper
```
Comment (tarebyte) [inline app/controllers/student_identifier_types_controller.rb]: Maybe instead of `:check_flipper` we can do something like `before_action :ensure_flipper_is_enabled`

/cc @johndbritton thoughts here?

File: app/views/organizations/_side_menu.html.erb [hunk 6]
```diff
@@ -1,7 +1,10 @@
 <nav class="menu">
   <span class="menu-heading">Classroom settings</span>
-  <%= link_to 'Classroom profile', edit_organization_path, class: "menu-item #{'selected' if current_page?(edit_organization_path)}" %>
+  <%= link_to 'Classroom profile', edit_organization_path(@organization), class: "menu-item #{'selected' if current_page?(edit_organization_path(@organization))}" %>
```
Comment (tarebyte) [inline app/views/organizations/_side_menu.html.erb]: This is fine for now, I'd love to come up with a reusable pattern for this.

File: app/views/student_identifier_types/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    Please do not ask students to provide any confidential information (e.g. SSN or bank account number) as their identifiers.
+  </div>
+  <%= form_for [@organization, @student_identifier_type] do |f| %>
+      <%= render partial: 'student_identifier_types/form_options', locals: { f: f } %>
+      <div class="form-actions">
+        <%= f.submit 'Create Identifier', class: 'btn btn-green js-navigation' %>
+        <%= link_to 'Cancel', session[:return_to], class: 'btn', role: 'button'%>
+      </div>
+  <% end %>
+</div>
```
Comment (tarebyte) [inline app/views/student_identifier_types/new.html.erb]: Can we fix the line ending here?

Thought rubocop checked for there

File: app/views/student_identifier_types/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+<div class="site-content-body">
+  <div class="flash flash-with-icon flash-form">
+    <%= octicon 'alert' %>
+    Please do not ask students to provide any confidential information (e.g. SSN or bank account number) as their identifiers.
+  </div>
+  <%= form_for [@organization, @student_identifier_type] do |f| %>
+      <%= render partial: 'student_identifier_types/form_options', locals: { f: f } %>
+      <div class="form-actions">
+        <%= f.submit 'Create Identifier', class: 'btn btn-green js-navigation' %>
+        <%= link_to 'Cancel', session[:return_to], class: 'btn', role: 'button'%>
```
Comment (tarebyte) [inline app/views/student_identifier_types/new.html.erb]: I think I see what you're getting at. Are there multiple places a teacher can click to create a student identifier?

File: config/routes.rb [hunk 9]
```diff
@@ -42,6 +42,7 @@
 
       resources :assignments
       resources :group_assignments, path: 'group-assignments'
+      resources :student_identifier_types, path: 'identifiers', only: [:index, :new, :create]
```
Comment (tarebyte) [inline config/routes.rb]: 👍

File: spec/controllers/student_identifier_types_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+  let(:organization)  { GitHubFactory.create_owner_classroom_org }
+  let(:user)          { organization.users.first                 }
+  let(:student)       { GitHubFactory.create_classroom_student   }
+
+  let(:student_identifier_type) { GitHubFactory.create_student_identifier(organization) }
+
+  before do
+    sign_in(user)
+    Classroom.flipper[:student_identifier].enable
```
Comment (tarebyte) [inline spec/controllers/student_identifier_types_controller_spec.rb]: Can we move this to another context.

```ruby
before do
  sign_in(user)
end

context "flipper is enabled for the user" do
  before do
    Classroom.flipper[:student_identifier].enable
  end
  ...
end

context "flipper is not enabled for the user" do
  ...
end
```

File: app/views/student_identifier_types/new.html.erb (hunk 8 above)
Comment (dinever) [inline app/views/student_identifier_types/new.html.erb]: @tarebyte Yep. A teacher can click into the page from the student identifier type management page or from the assignment creating page:



screen shot 2016-07-12 at 10 07 40 am

Comment (tarebyte) [inline app/views/student_identifier_types/new.html.erb]: Gotcha, also would you mind updating the button to use the Classroom green color?

Comment (tarebyte) [inline app/views/student_identifier_types/new.html.erb]: And remove the `+` icon

Comment (dinever) [inline app/views/student_identifier_types/new.html.erb]: :cool:, but this button will be in the next PR. I'll remember this.

File: app/views/organizations/_side_menu.html.erb (hunk 6 above)
Comment (dinever) [inline app/views/organizations/_side_menu.html.erb]: What about extracting menu items into another template?

in **organizations/_side_menu.html.erb**:

```
<%= render partial: 'organizations/menu_item', locals: { label: 'Classroom profile', path: edit_organization_path(@organization) } %>
```

in **organizations/_menu_item.html.erb**:

```
<%= link_to label, path, class: "menu-item #{'selected' if current_page?(path)}" %>
```

Comment (tarebyte) [inline app/views/organizations/_side_menu.html.erb]: I love that idea, let's do that in a different PR though.

**Comment by tarebyte** [comment]
:metal:
